plugins/modules/vmware_first_class_disk_info.py

```diff
@@ -0,0 +1,143 @@
+description:
+    - This module can be used to gather information about VMware vSphere First Class Disks.
+author:
+- Nina Loser (@nina2244)
+options:
+    datacenter_name:
+      description: The name of the datacenter.
+      type: str
+    datastore_name:
+      description: Name of datastore or datastore cluster of the disk.
+      required: true
+      type: str
+    disk_name:
+      description: The name of the disk. If not set return all found.
+      type: str
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Gather info of 1GBDisk
+  community.vmware.vmware_first_class_disk_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datastore_name: '{{ datastore_name }}'
+    disk_name: '1GBDisk'
+  register: disk_info
+  delegate_to: localhost
+
+- debug:
+    var: disk_info.first_class_disks
+
+- name: Gather info of all first class disks
+  community.vmware.vmware_first_class_disk_info:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    datastore_name: '{{ datastore_name }}'
+  register: disk_info
+  delegate_to: localhost
+
+- debug:
+    var: disk_info.first_class_disks
+'''
+
+RETURN = r'''
+first_class_disks:
+  description: list of dictionary of First-class disk and their information
+  returned: success
+  type: list
+  sample: [
+    {
+      "name": "1GBDisk",
+      "datastore_name": "DS0",
+      "size_mb": "1024",
+      "consumption_type": "disk",
+      "descriptor_version": 0,
+      "consumer_ids": []
+    }
+  ]
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.vmware.plugins.module_utils.vmware import PyVmomi, vmware_argument_spec
+
+
+class FirstClassDiskInfo(PyVmomi):
+    def __init__(self, module):
+        super(FirstClassDiskInfo, self).__init__(module)
+        self.datacenter_name = self.params['datacenter_name']
+        self.datastore_name = self.params['datastore_name']
+        self.disk_name = self.params['disk_name']
+
+    def gather_first_class_disk_info(self):
+        self.datastore_obj = self.find_datastore_by_name(datastore_name=self.datastore_name, datacenter_name=self.datacenter_name)
+        if not self.datastore_obj:
+            self.module.fail_json(msg='Failed to find datastore %s.' % self.datastore_name)
+
+        if self.disk_name:
+            self.disk_obj = self.find_first_class_disk_by_name(self.disk_name, self.datastore_obj)
+            if not self.disk_obj:
+                self.module.fail_json(msg='Failed to find disk %s.' % self.disk_name)
+
+            self.disks = [self.disk_obj]
+
+        else:
+            self.disks = self.find_first_class_disks(self.datastore_obj)
+            if not self.disks:
+                self.module.fail_json(msg='Failed to find any disk first class disk.')
```

Comment:
I don't understand why you want to fail here. If there are no FCDs, I think it would be better to just return an empty list. Or maybe `None`, but at the spur of the moment I would say an empty list `[]` would be best.

Reply:
@ihumster What do you think, `None` or `[]`? I tend to the empty list `[]`. But definitively not at a failure.

Reply:
@mariolenz I agree - empty list of disks is not at a failure. I think correct return empty list `[]`.

Reply:
Yes you are right. I will change it.